cmd/watch-termination/main.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+func main() {
+	terminationLog := flag.String("termination-log-file", "", "Write logs after SIGTERM to this file (in addition to stderr)")
+	terminationLock := flag.String("termination-touch-file", "", "Touch this file on SIGTERM and delete on termination")
+
+	flag.Parse()
+	args := flag.CommandLine.Args()
+
+	if len(args) == 0 {
+		fmt.Println("Missing command line")
+		os.Exit(1)
+	}
+
+	// use special tee-like writer when termination log is set
+	termCh := make(chan struct{})
+	var stderr io.Writer = os.Stderr
+	if len(*terminationLog) > 0 {
```

Comment:
is `terminationLog` a pointer ?

Reply:
yes. See flag library.

Reply:
nit: we could use https://golang.org/pkg/flag/#StringVar and pass the address of a local var

Reply:
not better. This is a common Golang pattern. It's the default flag library after all ;-)